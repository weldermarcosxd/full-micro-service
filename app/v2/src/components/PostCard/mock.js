export const PostCardMock = {
    post: {
        title: "title 1",
        body: "body 1",
        id: 1,
        cover: "https://in.ign.com/spider-man-ps4/127764/feature/should-you-buy-marvels-spider-man"
    }
};