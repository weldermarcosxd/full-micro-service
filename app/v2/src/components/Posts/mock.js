export const PostsMock = {
    posts: [{
        title: "title 1",
        body: "body 1",
        id: 1,
        cover: "https://in.ign.com/spider-man-ps4/127764/feature/should-you-buy-marvels-spider-man"
    },
    {
        title: "title 2",
        body: "body 2",
        id: 2,
        cover: "https://in.ign.com/spider-man-ps4/127764/feature/should-you-buy-marvels-spider-man"
    },
    {
        title: "title 3",
        body: "body 3",
        id: 3,
        cover: "https://in.ign.com/spider-man-ps4/127764/feature/should-you-buy-marvels-spider-man"
    }]
};