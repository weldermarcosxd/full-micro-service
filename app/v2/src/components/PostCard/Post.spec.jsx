import { screen, render } from "@testing-library/react";
import { PostCard } from ".";

import { PostCardMock } from "./mock";

const props = PostCardMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);
    const img = screen.getByRole("img", { name: /title 1/i });
    const heading = screen.getByRole("heading", { name: /title 1/i });
    const paragraph = screen.getByText("body 1");
    expect(img).toHaveAttribute("src", props.post.cover);
    expect(img).toBeInTheDocument();
    expect(heading).toBeInTheDocument();
    expect(paragraph).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
