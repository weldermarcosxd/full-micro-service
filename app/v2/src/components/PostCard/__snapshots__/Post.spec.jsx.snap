// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 1"
    src="https://in.ign.com/spider-man-ps4/127764/feature/should-you-buy-marvels-spider-man"
  />
  <div
    class="post-content"
  >
    <h2>
      title 1
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
